Rasa_Bot/actions/helper.py

```diff
@@ -95,8 +100,102 @@ def compute_preferred_time(tracker):
     return preferred_time
 
 
-def store_profile_creation_data_to_db(user_id: int, godin_activity_level: int,
-                                      running_walking_pref: int,
+def get_last_completed_dialog_part_from_db(user_id: int,
+                                           intervention_component_id: int):
+    """Get last completed dialog part from db."""
+    
+    session = get_db_session(db_url=DATABASE_URL)
+
+    selected = (
+        session.query(
+            UserInterventionState
+        ).order_by(UserInterventionState.last_time.desc())
+        .filter(
+            UserInterventionState.users_nicedayuid == user_id,
+            UserInterventionState.intervention_component_id == intervention_component_id
+        )
```

Comment:
in the table there is also the `completed` field that we can check to filter out the results.

```suggestion
            UserInterventionState.users_nicedayuid == user_id,
            UserInterventionState.intervention_component_id == intervention_component_id
            UserInterventionState.completed == False
```